src/components/SkillTopics.tsx

```diff
@@ -0,0 +1,87 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useContext } from "react";
+import { GetTopicsInfosQuery } from "../generated/graphql";
+import {
+  ADD_SKILL_TO_TOPIC,
+  DELETE_SKILL_TO_TOPIC,
+} from "../graphql/mutations/skills";
+import { GET_TOPICS_INFOS } from "../graphql/queries/topics";
+import { useDarkMode } from "../utils/darkMode";
+import { i18nContext } from "../utils/i18nContext";
+import { FetchedSkill } from "../utils/types";
+
+type skillTopicsProps = {
+  refetch: any;
+  readOnly: boolean;
+  skill: FetchedSkill;
+  skillTopics: any;
+};
+
+const SkillTopics = ({
+  refetch,
+  readOnly,
+  skill,
+  skillTopics,
+}: skillTopicsProps) => {
+  const { t } = useContext(i18nContext);
+  const { darkMode } = useDarkMode();
+  const { data: topics, loading: loadingTopics } =
+    useQuery<GetTopicsInfosQuery>(GET_TOPICS_INFOS);
+
+  const [insertTopic] = useMutation(ADD_SKILL_TO_TOPIC);
+  const [deleteTopic] = useMutation(DELETE_SKILL_TO_TOPIC);
+
+  const updateTopic = (selectedTopic: { id: string; name: string }) => {
+    const skilltopic = skillTopics.SkillTopic?.find(
+      (value) => value.topicId === selectedTopic?.id
+    );
+    if (!skilltopic) {
+      insertTopic({
+        variables: { skillId: skill?.id, topicId: selectedTopic.id },
+      }).then(() =>
+        refetch({
+          variables: { skillId: skill?.id },
+        })
+      );
+    } else {
+      deleteTopic({
+        variables: { skillId: skill?.id, topicId: selectedTopic.id },
+      }).then(() =>
+        refetch({
+          variables: { skillId: skill?.id },
+        })
+      );
+    }
+  };
+
+  return (
+    <div
+      className={`flex flex-col rounded-lg ${
+        darkMode
+          ? "bg-light-dark dark:bg-dark-dark my-2 p-2"
+          : "bg-light dark:bg-dark-dark my-2 p-2"
```

Comment:
you should put my-2, p-2 and dark:bg-dark-dark outside of the darkmode condition as you will always have it
also dark:bg-dark-dark in darkmode and lightmode ? seems weird or good behavior ?